tests/unit/test_timer.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+import pytest
+from reco_utils.common.timer import Timer
+
+
+@pytest.fixture(scope="function")
+def t():
+    return Timer()
+
+
+def test_no_time(t):
+    assert t.interval == 0
+    with Timer() as t2:
+        assert t2.interval == 0
+
+
+def test_stop_before_start(t):
+    with pytest.raises(ValueError):
+        t.stop()
+
+
+def test_timer(t):
+    big_num = 1000
+    t.start()
+    r = 0
+    a = [r + i for i in range(big_num)]
+    t.stop()
+    assert t.interval < 1
+    r = 0
+    with Timer() as t2:
+        a = [r + i for i in range(big_num)]
+    assert t2.interval < 1
```

Comment:
is this correct way to test? what about use time.time() or timeit's and compare the values between manual time measurement and our module with a small tolerance value?

Reply:
that's a good idea, I'm adding a solution with time.sleep(1).